### DISCUSSION TITLE
编译器是怎么实现`const`的？

### DISCUSSION THREAD

**Discussion by xuantengh** [liusy58/WhyThis#28] [General]
貌似没有专门的汇编指令去做这件事，编译器是检查声明周期里对特定内存区域的写操作嘛？

**Comment by howardlau1999** [comment]
确实是检查对于 const 变量，是否应用了 const 操作，也就是只有编译期的检查。不过，不同的虚拟内存段有读写权限的分别，如果编译器将对应的内存区域放到了只读虚拟内存段，运行时写会 Segmentation Fault
